src/predictor/gpu_predictor.cu

```diff
@@ -324,22 +336,31 @@ class GPUPredictor : public xgboost::Predictor {
                 h_nodes.begin() + h_tree_segments[tree_idx - tree_begin]);
     }
 
-    size_t i_batch = 0;
     size_t batch_offset = 0;
     for (auto &batch : dmat->GetRowBatches()) {
-      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
-      // out_preds have been sharded and resized in InitOutPredictions()
-      batch.offset.Shard(GPUDistribution::Overlap(devices_, 1));
+      GPUSet devices = GPUSet::All(param_.gpu_id, param_.n_gpus, batch.Size());
+      ConfigureShards(devices);
+      std::vector<size_t> out_preds_offsets;
+      PredictionDeviceOffsets(out_preds->Size(), batch_offset, batch.Size(),
+                              model.param.num_output_group, &out_preds_offsets);
+      out_preds->Reshard(GPUDistribution::Explicit(devices_, out_preds_offsets));
```

Comment:
i *think* this should consider the possibility that some shards may not end up getting any rows (0), and that code that relies on a non trivial number for the number of rows have to change to accommodate this possibility, which is what the other pr does.

Reply:
Added checks.